mock/py/mockbuild/buildroot.py

```diff
@@ -923,6 +924,7 @@ def delete(self):
             file_util.rmtree(self.basedir, selinux=self.selinux)
         self.chroot_was_initialized = False
         self.plugins.call_hooks('postclean')
+        self._unlock_buildroot()
```

Comment:
This move doesn't look bad, but I need an explanation of why you need to do this.  What is triggering the misbehavior, concrete example?

Reply:
If it is unlock _before_ the chroot has been completely cleaned (with rmtree), then other process might kick in and will see a chroot half in the way of being cleaned.  If the two mock processes are doing a clean first then they will start two rmtree at the same time, which will cause exceptions as seen in !1071. For instance, I just tried again moving it where it was and running two process in parallel gives this exception in one of them:
```
  File "/usr/lib/python3.11/site-packages/mockbuild/file_util.py", line 68, in rmtree
    os.rmdir(path)
OSError: [Errno 39] Directory not empty: '/var/lib/mock/fedora-36-x86_64/root/usr/share/man/ja'
```
Moving the unlock after the chroot is completely removed eliminates this kind of race condition.

Whether the unlock can happen _before_ or _after_ the call_hooks('postclean') is not clear to me. I thought that it might be safer to do it after.